apps/teamsjs-diagnostic-app/src/components/sample/CustomScenario.tsx

```diff
@@ -1,65 +1,94 @@
 import React, { useState } from 'react';
 import './CustomScenario.css';
-
-import apiComponents from './ApiComponents';
+import { useDrop } from 'react-dnd';
+import apiComponents, { ApiComponent } from './ApiComponents';
+import AppInstallDialogAPIs from '../../apis/AppInstallDialogApi';
+import BarCodeAPIs from '../../apis/BarCodeApi';
 
 const CustomScenario: React.FC = () => {
   const [searchQuery, setSearchQuery] = useState<string>('');
+  const [customScenario, setCustomScenario] = useState<Array<{ api: ApiComponent, func: string, input?: string }>>([]);
+
+  const handleRunScenario = async () => {
+    console.log('Running custom scenario...');
+    for (const { api, func, input } of customScenario) {
+      console.log(`Executing ${func} for ${api.title} with input: ${input}`);
+      // Execute the API function based on the selected function and input
+    }
+  };
+
+  const addToScenario = (api: ApiComponent, func: string, input?: string) => {
+    console.log(`Adding ${func} for ${api.title} with input: ${input}`);
+    setCustomScenario([...customScenario, { api, func, input }]);
+  };
+
+  const removeApiFromScenario = (index: number) => {
+    setCustomScenario(customScenario.filter((_, i) => i !== index));
+  };
+
+  const clearScenario = () => {
+    setCustomScenario([]);
+  };
+
+  const [{ isOver }, drop] = useDrop(() => ({
+    accept: 'API',
+    drop: (item: { api: ApiComponent, func: string, input?: string }) => addToScenario(item.api, item.func, item.input),
+    collect: (monitor) => ({
+      isOver: monitor.isOver(),
+    }),
+  }));
 
   const generateVerticalBoxes = () => {
     const filteredApis = apiComponents.filter(api =>
       api.title.toLowerCase().includes(searchQuery.toLowerCase())
     );
 
-    const options = ['Option 1', 'Option 2', 'Option 3'];
-    return filteredApis.map((api, index) => (
+    return filteredApis.map((api: ApiComponent, index: number) => (
       <div key={index} className="vertical-box">
-        <div className="api-container">
-          <div className="api-header">{api.title}</div>
-          <div className="dropdown-menu">
-            <label htmlFor={`select-${index}`} className="sr-only">
-              Select an option for API {index}
-            </label>
-            <select id={`select-${index}`} className="box-dropdown">
-              {options.map((option, optionIndex) => (
-                <option key={optionIndex} value={option}>
-                  {option}
-                </option>
-              ))}
-            </select>
-          </div>
-        </div>
+        {api.title === 'App Install Dialog API' ? (
+          <AppInstallDialogAPIs apiComponent={api} onDropToScenarioBox={addToScenario} />
```

Comment:
This is again a piece of feedback you choose to take or leave. It is a little brittle to have to check the api title field's literal string value here. That string is also specified in ApiComponents.tsx, and if you change it in one place you'd have to change it both or your app will break.

One solution you could use to improve this would be to add something like a factory function to the ApiComponent interface. You already have an "onClick" function, you could add another one that was "function that returns the right React Component when called". If you had something liek this, this section would just turn into:

```typescript
api.renderComponent(addToScenario)
```